### ISSUE TITLE
test on appveyor

### ISSUE THREAD

**Issue by niphlod** [web2py/pydal#117]
To my knowledge, appveyor is the only online service that has a Windows environment readily available, and it ships with several version of MSSQL installed.

I'd set up CI on that too, after a bit of trial and error I got that working (look at https://github.com/niphlod/pydal/tree/tests/appveyor). The rationale behind this is that I can't keep up with being the only man testing pydal commits on MSSQL, but I'd reaaaally like that tests continue to pass and that mssql is still kept on the "officially supported and battle-tested" backends. 
Of course integration with coveralls is problematic but until something proper comes up (travis-ci started to look into Windows environments more than a year ago and still has no ETA on it) at least we'd have MSSQL tested.
appveyor isn't bad at all, but of course isn't as fast as dockerized travis-ci.
Results are like https://ci.appveyor.com/project/niphlod/pydal/build/1.0.7 and badges are available, with notifications too.
What do you think ?

**Comment by gi0baro** [comment]
+1 on this. Unfortunately I don't see any github integration, but at least we have CI tests with this.

**Comment by niphlod** [comment]
I know. Fingers crossed but until then is surely better appveyor than my test rig :-P

**Comment by niphlod** [comment]
Moreover... since I'm exploiting all kinds of - really nice - services.... codecov.io instead of coveralls for coverage: it "coalesces" results from different CIs (for our purposes, travis-ci and appveyor), take a look at https://codecov.io/github/niphlod/pydal?ref=tests/appveyor . it shows coverage coming from 8 builds on travis-ci and 1 from appveyor...
bonus points for comments on PRs with graph, speed, suggestions, chrome plugin.
The only thing we'd miss is coverage history, but since pydal is fresh, IMHO it's not a big deal.
@mdipierro, @gi0baro  : thoughts ?

**Comment by gi0baro** [comment]
I agree with using appveyor and codecov.

**Comment by mdipierro** [comment]
I agree too

**Comment by niphlod** [comment]
will do in a few days.

**Comment by niphlod** [comment]
@mdipierro : I'll need 5 minutes of your time because there's the codecov repo key to encrypt to make codeveyor work, and I haven't write permissions on web2py/pydal

**Comment by gi0baro** [comment]
@niphlod I can do it as well

**Comment by niphlod** [comment]
one or the other is fine :P

**Comment by niphlod** [comment]
found a way to do it without requiring your intervention. I'd need https://github.com/web2py/pydal/pull/119 merged in before making the PR, though

**Comment by gi0baro** [comment]
Merged.

**Comment by niphlod** [comment]
please merge https://github.com/web2py/pydal/pull/120 in. 
It should enable everything properly. 
I'll make another PR fixing something else (e.g. the README to include the new badges) . That'll be used as a verification that everything is working smoothly for PRs too.

**Comment by gi0baro** [comment]
Closing this after #121 merge.
